doc/utils/manifest.md

```diff
@@ -47,6 +79,54 @@ This section is optional. It contains metadata used to run application tests.
 | `unit_directory`   | Path of the directory where unit tests can be found                                            |
 | `pytest_directory` | Path of the directory where functional, Python test can be found (`conftest.py` file expected) |
 
+#### `[tests.dependencies.<test_use_case>]`
+
+The tests.dependencies.* sections are optional. They contain a list of apps metadata helping building side applications needed for your tests.
+You can define as many as you need.
+The VSCode extension leverages this field to build test dependencies for Speculos tests or device tests.
+
+The syntax to describe a single side application is the following:
+
+| Field name | Description                                                           |
+|------------|-----------------------------------------------------------------------|
+| `url`      | The URL of the application git repository  |
+| `ref`      | The reference to checkout in the repository |
+| `use_case` | The `use_case` to use to build the application |
+
+This is to be included in a list of applications attached to a particular testing scenario, for instance:
+```toml
+[[tests.dependencies.test_use_case]]
+url = <url>
+ref = <ref>
+use_case = <use_case>
+```
+This specifies that the `test_use_case` scenario needs the application located at `<url>` on ref `<ref>` and should be built according to its `<use_case>`.
+
+This following syntax can be used, as it is lighter and equivalent to the previous one:
+```toml
+[tests.dependencies]
+test_use_case = [{url = <url>, ref = <ref>, use_case = <use_case>}]
+```
+
+> [!INFO]
+> Do note that the `<use_case>` is the one of the application referred to, not to this manifest's application. Hence it must be defined on the other application own manifest (or be `"default"`)
+
+> [!INFO]
+> The `<use_case>` is optional, and will fallback to `use_case = "default"` if non present.
+
+Example for an Ethereum plugin that needs the Ethereum application sideloaded on the device:
+```toml
+[tests.dependencies] # Set of applications to load on the device (or Speculos) for a given test suite
+
+testing_with_prod = [
+  {url = "https://github.com/LedgerHQ/app-ethereum", ref = "master", use_case = "default"},
+]
+
+testing_with_latest = [
+  {url = "https://github.com/LedgerHQ/app-ethereum", ref = "develop", use_case = "default"},
```

Comment:
Maybe we recommend testing with debug? So that by default user that do this will have logs in ragger output?

Reply:
Changed to use `cal_key_debug`